dask/groupby-dask2.py

```diff
@@ -189,147 +191,166 @@
+print(ans.head(3), flush=True)
+print(ans.tail(3), flush=True)
+del ans
 
-#question = "sum v3 count by id1:id6" # q10
-#gc.collect()
-#t_start = timeit.default_timer()
-#ans = x.groupby(['id1','id2','id3','id4','id5','id6'], dropna=False, observed=True).agg({'v3':'sum', 'v1':'size'}).compute() # column name different than expected, ignore it because: ValueError: Metadata inference failed in `rename`: Original error is below: ValueError('Level values must be unique: [nan, nan] on level 0',)
-#ans.reset_index(inplace=True)
-#print(ans.shape, flush=True)
-#t = timeit.default_timer() - t_start
-#m = memory_usage()
-#t_start = timeit.default_timer()
-#chk = [ans.v3.sum(), ans.v1.sum()]
-#chkt = timeit.default_timer() - t_start
-#write_log(task=task, data=data_name, in_rows=in_rows, question=question, out_rows=ans.shape[0], out_cols=ans.shape[1], solution=solution, version=ver, git=git, fun=fun, run=1, time_sec=t, mem_gb=m, cache=cache, chk=make_chk(chk), chk_time_sec=chkt, on_disk=on_disk)
-#del ans
-#gc.collect()
-#t_start = timeit.default_timer()
-#ans = x.groupby(['id1','id2','id3','id4','id5','id6'], dropna=False, observed=True).agg({'v3':'sum', 'v1':'size'}).compute()
-#ans.reset_index(inplace=True)
-#print(ans.shape, flush=True)
-#t = timeit.default_timer() - t_start
-#m = memory_usage()
-#t_start = timeit.default_timer()
-#chk = [ans.v3.sum(), ans.v1.sum()]
-#chkt = timeit.default_timer() - t_start
-#write_log(task=task, data=data_name, in_rows=in_rows, question=question, out_rows=ans.shape[0], out_cols=ans.shape[1], solution=solution, version=ver, git=git, fun=fun, run=2, time_sec=t, mem_gb=m, cache=cache, chk=make_chk(chk), chk_time_sec=chkt, on_disk=on_disk)
-#print(ans.head(3), flush=True)
-#print(ans.tail(3), flush=True)
-#del ans
+question = "sum v3 count by id1:id6"  # q10
+print(question)
+gc.collect()
+t_start = timeit.default_timer()
+ans = (
+  x.groupby(
+    ['id1', 'id2', 'id3', 'id4', 'id5', 'id6'],
+    dropna=False,
+    observed=True,
+  )
+  .agg({'v3': 'sum', 'v1': 'size'}, split_out=x.npartitions)
+  .rename(columns={"v1": "count"})
+  .compute()
+)
+ans.reset_index(inplace=True)
+print(ans.shape, flush=True)
+t = timeit.default_timer() - t_start
+m = memory_usage()
+t_start = timeit.default_timer()
+chk = [ans.v3.sum(), ans["count"].sum()]
+chkt = timeit.default_timer() - t_start
+write_log(task=task, data=data_name, in_rows=in_rows, question=question, out_rows=ans.shape[0], out_cols=ans.shape[1], solution=solution, version=ver, git=git, fun=fun, run=1, time_sec=t, mem_gb=m, cache=cache, chk=make_chk(chk), chk_time_sec=chkt, on_disk=on_disk)
+del ans
+gc.collect()
+t_start = timeit.default_timer()
+ans = (
+  x.groupby(
+    ['id1', 'id2', 'id3', 'id4', 'id5', 'id6'],
+    dropna=False,
+    observed=True,
+  )
+  .agg({'v3': 'sum', 'v1': 'size'}, split_out=x.npartitions)
```

Comment:
The only thing I added to this is this line
`split_out=x.npartitions`

This forces dask to not reduce the result immediately onto a single worker since our workers operate on strict memory boundaries and the reduction would kill them. Reducing the final result at the very end should be doable (this query actually doesn't reduce data a lot but scheduler and client are not artificially memory bound)